python-avd/pyavd/_eos_designs/structured_config/underlay/port_channel_interfaces.py

```diff
@@ -113,6 +114,47 @@ def port_channel_interfaces(self: AvdStructuredConfigUnderlay) -> list | None:
 
             port_channel_interfaces.append(port_channel_interface)
 
+        # Support l3_port_channels including sub-interfaces
+        subif_parent_port_channel_names = set()
+        regular_l3_port_channel_names = set()
+        for l3_port_channel in self.shared_utils.node_config.l3_port_channels:
+            interface_name = l3_port_channel.name
+            if "." in interface_name:
+                # This is a subinterface for a port-channel interface.
+                # We need to ensure that parent port-channel interface is also included explicitly
+                # within list of Port-Channel interfaces.
+                parent_port_channel_name = interface_name.split(".", maxsplit=1)[0]
+                subif_parent_port_channel_names.add(parent_port_channel_name)
+                # TODO: Unable to add validation for 'mode' setting for Port-Channel sub-interface.
+                # Since we have default value specified in schema for this,
+                # we end up finding default value even when no explicit value is specified.
+                if l3_port_channel.member_interfaces:
+                    msg = f"Port-Channel sub-interface '{l3_port_channel}' has 'member_interfaces' set.This is not a valid setting."
+                    raise AristaAvdInvalidInputsError(msg)
+            else:
+                # This is a regular Port-Channel (not sub-interface)
+                regular_l3_port_channel_names.add(interface_name)
+
+        # Sanity check if the parent Port-channel for sub-interface is specified
+        for parent_port_channel in subif_parent_port_channel_names:
+            if parent_port_channel not in regular_l3_port_channel_names:
+                msg = "At least one L3 Port-Channel subinterface does not have parent Port-Channel interface specified."
+                raise AristaAvdInvalidInputsError(msg)
```

Comment:
Wondering if we should not try to be more specifc here by listing all of the ones with issues using set difference:

+ refactor error message

```suggestion
        if (missing_parent_port_channel := subif_parent_port_channel_names.difference(regular_l3_port_channel_names)):
                msg = "At least one L3 Port-Channel subinterface does not have parent Port-Channel interface specified."
                raise AristaAvdInvalidInputsError(msg)
```

Reply:
Thank you for the hint to use difference() on set
Added the same and revised validation error message to include all missing Port-Channels (parent of sub-interfaces)
Added negative unit test `missing-parent-of-l3-port-channel-sub-interface.yml` for the same